packages/lib/server/repository/eventType.ts

```diff
@@ -192,11 +207,97 @@ export class EventTypeRepository {
           ...where,
         },
         select,
+        cursor: cursor ? { id: cursor } : undefined,
+        take: limit ? limit + 1 : undefined, // We take +1 as itll be used for the next cursor
```

Comment:
We could remove the duplicates at 3 places and compute cursor and take at a single place at the top.